cpp/ql/src/experimental/Security/CWE/CWE-193/InvalidPointerDeref.ql

```diff
@@ -179,6 +179,22 @@ predicate isSinkImpl(
   pointerAddInstructionHasBounds(pai, sink1, sink2, delta)
 }
 
+/**
+ * Yields any instruction that is control-flow reachable from `instr`.
+ */
+Instruction getASuccessor(Instruction instr) {
```

Comment:
This may need to be inlined since "every instruction that is reachable from a given instruction" is a _very_ large relation. But let's see what DCA says 😄.

Reply:
Yeah, this needs to be inlined. I cancelled it after a while at which point I had:
```
[2023-06-02 13:28:17] (1459s) Tuple counts for InvalidPointerDeref#e13af12b::getASuccessor#1#ff/2@a5e22c71 after 14m58s:
                      72359368   ~0%     {2} r1 = JOIN Instruction#577b6a83::Instruction::getBlock#0#dispred#ff_10#join_rhs WITH boundedFastTC:IRBlock#896e97af::Cached::blockSuccessor#2#ff_10#higher_order_body:Instruction#577b6a83::Instruction::getBlock#0#dispred#ff_1#higher_order_body ON FIRST 1 OUTPUT Rhs.1, Lhs.1 'result'
                      1801630803 ~0%     {2} r2 = JOIN r1 WITH Instruction#577b6a83::Instruction::getBlock#0#dispred#ff_10#join_rhs ON FIRST 1 OUTPUT Rhs.1 'result', Lhs.1 'result'
                      
                      0          ~0%     {3} r3 = JOIN Instruction#577b6a83::Instruction::getBlock#0#dispred#ff_10#join_rhs WITH IRBlock#896e97af::Cached::getInstruction#2#fff_021#join_rhs ON FIRST 2 OUTPUT Lhs.0, Lhs.1 'result', Rhs.2
                      0          ~0%     {4} r4 = JOIN r3 WITH Instruction#577b6a83::Instruction::getBlock#0#dispred#ff_10#join_rhs ON FIRST 1 OUTPUT Lhs.0, Rhs.1 'result', Lhs.1 'result', Lhs.2
                      0          ~0%     {4} r5 = JOIN r4 WITH IRBlock#896e97af::Cached::getInstruction#2#fff_021#join_rhs ON FIRST 2 OUTPUT Lhs.2 'result', Lhs.3, Lhs.1 'instr', Rhs.2
                      0          ~0%     {4} r6 = SELECT r5 ON In.1 >= In.3
                      0          ~0%     {2} r7 = SCAN r6 OUTPUT In.2 'instr', In.0 'result'
                      
                      1801630803 ~0%     {2} r8 = r2 UNION r7
                                         return r8
```

Reply:
Should be resolved by https://github.com/github/codeql/pull/13355/commits/8ac1d56a7f880ad5348b93765db0e83a0a17a7be